src/prefect/transactions.py

```diff
@@ -148,8 +162,13 @@ def __exit__(self, *exc_info):
         self.reset()
 
     def begin(self):
-        # currently we only support READ_COMMITTED isolation
-        # i.e., no locking behavior
+        if (
+            self.store
+            and self.key
+            and self.isolation_level == IsolationLevel.SERIALIZABLE
+        ):
+            self.logger.debug(f"Acquiring lock for transaction {self.key!r}")
+            self.store.acquire_lock(self.key)
```

Comment:
Any thought to whether a transaction should have its own ID so that we can have a more specific `holder` here?  Or can there only ever be one transaction active on a thread at a time?

Reply:
If each transaction had a unique ID, you could cause a deadlock by opening a child transaction with the same key. It's an edge case, but I lean toward avoiding the deadlock in that situation, but I could be convinced otherwise.

Reply:
Ahh that's a good catch.  Since `Transaction` is a `ContextModel` I think there should only be one active on a thread at a time, right?

Reply:
There's one transaction context active at a time (i.e., `get_transaction` will only return the active transaction based on the context), but multiple transactions can be open simultaneously. In normal operations, those transactions would have different keys, be written in different records, and not conflict with each other.